### DISCUSSION TITLE
Missing symbols when linking raylib in Zig

### DISCUSSION THREAD

**Discussion by sunbeam60** [raysan5/raylib#3932] [Q&A]
Hi,

I'm getting missing symbols when linking a statically built raylib into a Zig program:
```
error: lld-link: undefined symbol: timeBeginPeriod
error: lld-link: undefined symbol: timeEndPeriod
error: lld-link: undefined symbol: __declspec(dllimport) CreateDCW
error: lld-link: undefined symbol: __declspec(dllimport) GetDeviceGammaRamp
error: lld-link: undefined symbol: __declspec(dllimport) DeleteDC
error: lld-link: undefined symbol: __declspec(dllimport) SetDeviceGammaRamp
error: lld-link: undefined symbol: __declspec(dllimport) GetDeviceCaps
error: lld-link: undefined symbol: __declspec(dllimport) ChoosePixelFormat
error: lld-link: undefined symbol: __declspec(dllimport) SetPixelFormat
error: lld-link: undefined symbol: __declspec(dllimport) DescribePixelFormat
error: lld-link: undefined symbol: __declspec(dllimport) CreateDIBSection
error: lld-link: undefined symbol: __declspec(dllimport) CreateBitmap
error: lld-link: undefined symbol: __declspec(dllimport) DeleteObject
error: lld-link: undefined symbol: __declspec(dllimport) SwapBuffers
error: lld-link: undefined symbol: __declspec(dllimport) CreateRectRgn

```
I was under the impression (from https://github.com/raysan5/raylib/tree/master/src/external/glfw) that all GLFW dependencies were built as part of the library.

This could very well be a Zig issue (raylib was built using build.zig in the raylib distribution) but I'm a bit puzzled over the dllimport given I'm using the static lib of raylib (which should have GLFW inside it).

Anyone have any ideas?

**Comment by raysan5** [comment]
@sunbeam60 the missing symbols belong to Windows system libraries: `winmm` and `gdi32`
Those libraries are required on Windows build.

**Comment by sunbeam60** [comment]
Argh, as it politely lists here: https://github.com/raysan5/raylib/wiki/Working-on-Windows

Thanks for the rapid response.
